### DISCUSSION TITLE
code-server armhf release always missing some scripts on Android

### DISCUSSION THREAD

**Discussion by AnonymousM21** [coder/code-server#6424] [Q&A]
code-server works successfully on my samsung galaxy a11 android device. But the log output shows some files are missing always when I run it. I checked four versions of code-server, but it's the same always. What's is the solution?  Here is the log output: 

File not found: /usr/lib/code-server/lib/vscode/out/vsda_bg.wasm
File not found: /usr/lib/code-server/lib/vscode/out/vsda.js
[19:28:58] [127.0.0.1][7c7d4a64][ManagementConnection] New connection established.
[19:29:01] Using the in-memory credential store as the operating system's credential store could not be accessed. Please see https://aka.ms/vscode-server-keyring on how to set this up. Details: Cannot find module '../build/Release/keytar.node'
Require stack:
- /usr/lib/code-server/lib/vscode/node_modules/keytar/lib/keytar.js
- /usr/lib/code-server/lib/vscode/out/bootstrap-amd.js
- /usr/lib/code-server/out/node/util.js
- /usr/lib/code-server/out/node/cli.js
- /usr/lib/code-server/out/node/entry.js

A screeshot from Termux:

![Screenshot_20230910-193029_Termux](https://github.com/coder/code-server/assets/118679960/6e33b1b8-ecb8-4346-8efd-d29187650639)

A screenshot to show that it works successfully on my browser:

![Screenshot_20230910-193018_Kiwi Browser](https://github.com/coder/code-server/assets/118679960/7b65c557-f205-4ffd-b98d-c03a702d47b6)

Thanks,
Melaku

**Comment by lg-nineoneninethreenine** [comment]
I believe you use something else with Termux, and what I think has occured is that you use a VM along with Termux, and it doesn't have all the dependency files for you to run code-server. Can you give us more info on the way you use code-server with Termux in this case?

**Reply by AnonymousM21** [reply]
I used proot to install ubuntu on termux(without root). That's it!
Here are some error occured directories looks:
![Screenshot_20230910-205702_Termux](https://github.com/coder/code-server/assets/118679960/091a6655-9432-47c6-a3e6-0f57261509e5)


![20230910_205613](https://github.com/coder/code-server/assets/118679960/5be183f7-118a-46ec-a834-e26d38077c99)


![Screenshot_20230910-205057_Termux](https://github.com/coder/code-server/assets/118679960/248376fb-16b5-4b96-8a39-7737b068f678)


![Screenshot_20230910-205430_Termux](https://github.com/coder/code-server/assets/118679960/383aaf44-029c-424f-9d62-8b2e5b983402)

**Comment by code-asher** [comment]
You can ignore the `vsda` files, they are not available in OSS builds of VS Code.  They seem to be optional.

If keytar is missing it might be because the dependencies required to build it are missing.  I think it is `libsecret` or something like that.  But it is probably not worth getting it built because secret storage is broken right now anyway, using keytar requires dbus which is tricky, and it seems VS Code is moving away from keytar.

**Reply by AnonymousM21** [reply]
I tried to install `libsecret` and `keytar` even globally, but the error is the same. 
Here are the commands I've used:
```bash
$ sudo apt-get install pkg-config
$ sudo apt-get install libsecret-1-dev
`export PKG_CONFIG_PATH="/usr/lib/pkgconfig:$PKG_CONFIG_PATH"` to `~/.bashrc` file
$ npm install -g keytar
```

**Reply by code-asher** [reply]
If keytar continues to fail building it might just not be supported on Android, I am not sure.  Might have to ask the keytar folks.

But since secret storage is broken anyway even if you manage to set up dbus you might be better off just ignoring keytar for now.  I think it will be replaced in a future update.

If you need secret storage now one workaround is to replace keytar: https://github.com/coder/code-server/issues/5072#issuecomment-1191860722

**Reply by AnonymousM21** [reply]
As you suggest, I run:
```bash
$ curl -L https://raw.githubusercontent.com/stevenlafl/node-keytar/master/lib/keytar.js -o /usr/lib/code-server/lib/vscode/node_modules/keytar/lib/keytar.js
```
but it's not compatible for android, generated more errors than before.

Thanks you all for your replies! I'll wait for future updates of `code-server`, Keep going!

And the other bug that I've discovered on android is that when you run `code-server` with specifed directory like:
```bash
$ code-server --auth none ./home/m21/demoProject
```
it always opens the most recent window(project) instead of the specified directory, maybe consider it also in future updates.

Thanks again!

**Reply by code-asher** [reply]
Ah good to know, I have never tried the workaround  myself so I have no idea why it would not work on Android.

Hmm that is strange, code-server should prefer the directory passed on the command line over the most recent.  https://github.com/coder/code-server/blob/03dc8cd808fd27aa449ab9c4387cf130574003f0/docs/FAQ.md#L221-L228

Could be a bug.  Definitely feel free to open an issue.

**Reply by AnonymousM21** [reply]
Okay Thank you!
